examples/numalogic-simple-pipeline/ml_steps/udf/inference.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os
+
+import numpy as np
+from numalogic.models.autoencoder import AutoencoderPipeline
+from numalogic.models.autoencoder.variants import VanillaAE
+from pynumaflow.function import Messages, Message, Datum
+
+from ml_steps.utility import Payload, load_model
+
+LOGGER = logging.getLogger(__name__)
+DEFAULT_WIN_SIZE = 12
+WIN_SIZE = int(os.getenv("WIN_SIZE", DEFAULT_WIN_SIZE))
+
+
+def inference(key: str, datum: Datum) -> Messages:
+    # Load data and convert bytes to MessagePacket
+    payload = Payload.from_json(datum.value.decode("utf-8"))
+
+    messages = Messages()
+
+    artifact = load_model()
```

Comment:
DO train if artifact is None. We don't need to rely on a later exception to make that decision